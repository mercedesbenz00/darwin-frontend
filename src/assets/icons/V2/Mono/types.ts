export enum IconVariant {
  DEFAULT = 'default',
  DISABLED = 'disabled',
  SELECTED = 'selected',
  NEGATIVE = 'negative',
}
