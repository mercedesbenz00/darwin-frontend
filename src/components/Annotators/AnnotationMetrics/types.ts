import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
