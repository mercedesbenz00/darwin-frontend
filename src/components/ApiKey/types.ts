export type AbilityGroup =
  'manage_comments' |
  'manage_tasks' |
  'manage_datasets' |
  'manage_team' |
  'manage_team_members' |
  'manage_models'

export type Group = { id: AbilityGroup; name: string }
export type GroupOption = Group & { selected: boolean }

export enum SupportedLanguages {
  shell = 'shell',
  python = 'python',
  javascript = 'javascript',
  elixir = 'elixir',
  cli = 'cli'
}

export type AccessCodeExample = { language: SupportedLanguages, code: string }
