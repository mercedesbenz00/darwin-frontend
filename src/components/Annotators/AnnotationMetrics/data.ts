import { ChartDimension } from './types'

export const dimensionLabels: { [k in ChartDimension]: string } = {
  annotationTime: 'Time spent annotating',
  avgTimePerAnnotation: 'Avg seconds per annotation',
  imagesAnnotated: 'Total images annotated',
  annotationsCreated: 'Total annotations created',
  reviewPassRate: 'Review pass rate'
}
