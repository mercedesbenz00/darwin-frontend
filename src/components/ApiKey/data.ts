import { AbilityID } from '@/store/modules/apiKey/types'

import { AbilityGroup, Group } from './types'

/**
 * Defines groups to group user permissions by
 *
 * Definition order determines render order
 */
const PERMISSION_GROUPS: Group[] = [
  { name: 'Manage Team', id: 'manage_team' },
  { name: 'Manage Datasets', id: 'manage_datasets' },
  { name: 'Manage Tasks', id: 'manage_tasks' },
  { name: 'Manage Comments', id: 'manage_comments' },
  { name: 'Manage Team Members', id: 'manage_team_members' },
  { name: 'Manage Models', id: 'manage_models' }
]

/**
 * Defines to which group each ability/permission belongs to.
 *
 * If an ability is not listed in a group, it wont be rendered on the frontend.
 *
 * This means it's a backend-only ability and it's API key use is not supported.
 */
const GROUPED_ABILITIES: { [k in AbilityGroup]: AbilityID[] } = {
  manage_tasks: [
    'assign_items',
    'view_stage',
    'update_stage'

  ],
  manage_datasets: [
    'view_datasets',
    'export_dataset',
    'create_dataset',
    'fork_dataset',
    'update_dataset',
    'archive_dataset',
    'update_dataset_data',
    'archive_dataset_items',
    'delete_dataset_items',
    'view_annotations',
    'view_annotation_report',
    'view_dataset_annotators',
    'view_dataset_exports',
    'view_dataset_performance_report',
    'view_dataset_report',
    'view_annotation_classes',
    'create_annotation_class',
    'update_annotation_class',
    'delete_annotation_class',
    'import_annotations'
  ],
  manage_team: [
    'archive_team',
    'create_team',
    'transfer_team_ownership',
    'manage_customer',
    'update_team',
    'view_team'
  ],
  manage_team_members: [
    'delete_membership',
    'manage_admin_invitations',
    'manage_invitations',
    'update_membership',
    'view_invitations',
    'view_team_members'
  ],
  manage_comments: [
    'create_comment_thread',
    'create_comment',
    'delete_comment_thread',
    'delete_comment',
    'update_comment_thread',
    'update_comment',
    'view_comment_threads'
  ],
  manage_models: [
    'deploy_model',
    'train_models',
    'view_models',
    'run_inference'
  ]
}

/**
 * Lists detailed information about abilities, for friendlier rendering.
 *
 * - `name` is shown as the primary way to depict an ability.
 * - `info` is extended explanation of an ability, usually rendered on hover
 */
const ABILITY_INFO: {[k in AbilityID]: { name: string, info: string }} = {
  /* eslint-disable max-len */
  archive_dataset_items: { name: 'Archive and restore items', info: 'Archive and restore items in a dataset.' },
  archive_dataset: { name: 'Archive datasets', info: 'Archive the entire dataset.' },
  archive_team: { name: 'Archive team', info: 'Archive the entire team.' },
  assign_items: { name: 'Assign items', info: 'Assign items as workflows to annotators.' },
  create_annotation_class: { name: 'Create annotation classes', info: 'Create new annotation classes.' },
  create_comment_thread: { name: 'Start new comment threads.', info: 'Start new comment threads on an image, during annotation or review.' },
  create_comment: { name: 'Reply to comment threads', info: 'Post replies to existing comment threads.' },
  create_dataset: { name: 'Create datasets', info: 'Create new datasets.' },
  create_team: { name: 'Create teams', info: 'Create new teams.' },
  delete_annotation_class: { name: 'Delete annotation classes', info: 'Delete existing annotation classes.' },
  delete_comment_thread: { name: 'Delete comment threads', info: 'Delete existing comment threads, during annotation or review.' },
  delete_comment: { name: 'Delete comments', info: 'Delete comments in existing comment threads.' },
  delete_dataset_items: { name: 'Delete dataset items', info: 'Permanently delete dataset items.' },
  delete_membership: { name: 'Delete memberships', info: 'Permanently remove users from the team.' },
  deploy_model: { name: 'Deploy existing neural models', info: 'Deploy existing neural models.' },
  export_dataset: { name: 'Export datasets', info: 'Export annotation data from datasets.' },
  fork_dataset: { name: 'Fork datasets', info: 'Fork existing datasets to create new ones.' },
  import_annotations: { name: 'Import annotations', info: 'Import annotations from external sources' },
  manage_admin_invitations: { name: 'Create admin invites', info: 'Invite users to become team administrators.' },
  manage_customer: { name: 'View and update billing information', info: 'View and update your billing information.' },
  manage_invitations: { name: 'Create invites', info: 'Invite users to become team members or annotators.' },
  run_inference: { name: 'Run model inference', info: 'Use special tools based on deployed models to annotate images.' },
  tag_dataset_images: { name: 'Tag dataset images', info: 'Tag entire images with classes.' },
  train_models: { name: 'Train new neural models', info: 'Train new neural models.' },
  transfer_team_ownership: { name: 'Transfer team ownership', info: 'Transfer ownership of this team to another user.' },
  update_annotation_class: { name: 'Update annotation classes', info: 'Update existing annotation classes.' },
  update_comment_thread: { name: 'Edit comment threads', info: 'Edit content of comment threads.' },
  update_comment: { name: 'Edit comments', info: 'Edit content of comments.' },
  update_dataset_data: { name: 'Add data to datasets', info: 'Add images, videos or any other supported data to datasets.' },
  update_dataset: { name: 'Update datasets', info: 'Edit dataset information.' },
  update_membership: { name: 'Update team members', info: 'Change a team members role.' },
  update_stage: { name: 'Update item workflow', info: 'Progress item through workflow stages' },
  update_team: { name: 'Update team', info: 'Change team information.' },
  view_annotation_classes: { name: 'View annotation classes', info: 'View annotation classes defined for a dataset.' },
  view_annotation_report: { name: 'View annotation reports', info: 'View or download annotation reports.' },
  view_annotations: { name: 'View annotations', info: 'View annotation data for an image.' },
  view_comment_threads: { name: 'View comment threads', info: 'Read comment threads created for an image.' },
  view_dataset_annotators: { name: 'View dataset annotators', info: 'View users marked as annotators for a specific dataset.' },
  view_dataset_exports: { name: 'View exported datasets', info: 'List exported dataset versions for downloading.' },
  view_dataset_performance_report: { name: 'View dataset performance reports', info: 'View performance reports for a dataset.' },
  view_dataset_report: { name: 'View dataset reports', info: 'View statistical reports for a dataset.' },
  view_datasets: { name: 'View datasets', info: 'View dataset information.' },
  view_invitations: { name: 'View invitations', info: 'View team invitations.' },
  view_models: { name: 'View neural models', info: 'View information on team neural models.' },
  view_stage: { name: 'View workflow stages', info: "View information about an item's workflow stages." },
  view_team_members: { name: 'View team members', info: 'View all team members.' },
  view_team: { name: 'View team', info: 'View team information.' }
  /* eslint-enable max-len */
}

export { ABILITY_INFO, GROUPED_ABILITIES, PERMISSION_GROUPS }
