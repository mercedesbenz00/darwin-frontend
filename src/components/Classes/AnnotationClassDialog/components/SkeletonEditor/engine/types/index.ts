export * from './SkeletonEdgeType'
export * from './SkeletonNodeType'
