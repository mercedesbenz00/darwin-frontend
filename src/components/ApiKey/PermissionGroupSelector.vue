<template>
  <div
    :key="group.id"
    class="permission-group"
    :class="{'permission-group--expanded': expanded}"
  >
    <div class="permission-group__header">
      <check-box
        :id="group.id"
        v-tooltip="'Click to toggle all actions in this group'"
        class="permission-group__header__control"
        :label="group.name"
        :name="group.id"
        :value="selected"
        @change="toggleGroup(group.id, $event.checked)"
      />
      <button
        class="permission-group__header__collapse-expand"
        @click="expanded = !expanded"
      >
        <div class="permission-group__header__collapse-expand__text">
          {{ outOf }}
        </div>
        <img
          class="permission-group__header__collapse-expand__icon"
          src="/static/imgs/chevron_bottom.svg"
        >
      </button>
    </div>
    <transition name="toggle">
      <ul
        v-if="expanded"
        class="permission-group__abilities"
      >
        <li
          v-for="ability in abilityOptions"
          :key="ability.id"
          v-tooltip="ability.info"
          class="permission-group__abilities__ability"
        >
          <check-box
            :id="ability.id"
            :label="ability.name"
            :name="ability.id"
            :value="ability.selected"
            @change="toggleAbility(ability.id, $event.checked)"
          />
        </li>
      </ul>
    </transition>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import { State } from 'vuex-class'

import CheckBox from '@/components/Common/CheckBox/V1/CheckBox.vue'
import { AbilityID } from '@/store/modules/apiKey/types'
import { Ability } from '@/store/types'

import { GROUPED_ABILITIES, ABILITY_INFO } from './data'
import { Group } from './types'

type NormalizedAbility = {id: string, name: string }
type AbilityOption = NormalizedAbility & { selected: boolean }

/**
 * Renders a permission group within the generalized key creation UI
 *
 * Has an expanded and collapsed state.
 *
 * The collapsed state allows selection of the entire group of permissions.
 * The expanded state allows selection of individual permissions.
 */
@Component({ name: 'permission-group-selector', components: { CheckBox } })
export default class PermissionGroupSelector extends Vue {
  @Prop({ required: true, type: Object })
  group!: Group

  @State(state => state.auth.abilities)
  userAbilities!: Ability[]

  expanded: boolean = false

  selectedAbilities: string[] = []

  get userActions (): AbilityID[] {
    const actions: AbilityID[] = this.userAbilities
      .reduce((acc, a) => acc.concat(a.actions as AbilityID[]), [] as AbilityID[])

    return GROUPED_ABILITIES[this.group.id].filter(a => actions.includes(a))
  }

  get abilities (): NormalizedAbility[] {
    return this.userActions.map(action => ({
      id: action,
      name: ABILITY_INFO[action].name,
      info: ABILITY_INFO[action].info
    }))
  }

  get abilityOptions (): AbilityOption[] {
    return this.abilities.map(a => ({ ...a, selected: this.selectedAbilities.includes(a.id) }))
  }

  toggleGroup (groupId: string, selected: boolean) {
    if (selected) {
      this.selectedAbilities = this.abilities.map(a => a.id)
    } else {
      this.selectedAbilities = []
    }

    this.$emit('change', this.group.id, this.selectedAbilities)
  }

  toggleAbility (id: string, selected: boolean) {
    const index = this.selectedAbilities.indexOf(id)
    if (selected && index === -1) { this.selectedAbilities.push(id) }
    if (!selected && index >= 0) { this.selectedAbilities.splice(index, 1) }

    this.$emit('change', this.group.id, this.selectedAbilities)
  }

  get selected (): boolean {
    return this.abilityOptions.every(o => o.selected)
  }

  get outOf (): string {
    const selectedCount = this.abilityOptions.filter(o => o.selected).length
    const totalCount = this.abilityOptions.length
    return `${selectedCount} / ${totalCount} selected`
  }
}
</script>

<style lang="scss" scoped>
.permission-group__header {
  @include row;
  align-items: center;
  justify-content: space-between;
  padding-bottom: 10px;
  box-sizing: border-box;
}

.permission-group--expanded .permission-group__header {
  border-bottom: 2px solid $colorPrimaryLight3;
}

.permission-group__abilities {
  padding-left: 20px;
  margin-top: 10px;
  column-count: 2;
  column-gap: 25px;
}

.permission-group__abilities__ability {
  margin-bottom: 10px;
  list-style-type: none;
}

.permission-group__header__control.check-box {
  width: auto;
  color: $colorSecondaryLight;
  margin-right: 15px;
}

.permission-group__header__collapse-expand {
  background: none;
  @include row;
  align-items: center;
  margin-right: 20px;
}

.permission-group__header__collapse-expand__text {
  color: $colorSecondaryLight;
  margin-right: 15px;
  flex: 1;
  text-align: right;
}

.permission-group__header__collapse-expand__icon {
  border-radius: 50%;
  padding: 5px;
  width: 25px;
  height: 25px;

  transition: transform .5s ease;
  transition: background-color .2s ease;
}

.permission-group--expanded .permission-group__header__collapse-expand__icon {
  transform: rotate(-180deg)
}

.permission-group__header__collapse-expand:hover .permission-group__header__collapse-expand__icon {
  background: $colorSecondaryLight2;
}

.toggle-enter-to,
.toggle-leave {
  max-height: 500px;
}

.toggle-enter-active,
.toggle-leave-active {
  transition: max-height .2s ease;
}

.toggle-enter,
.toggle-leave-to {
  max-height: 0px;
}
</style>
