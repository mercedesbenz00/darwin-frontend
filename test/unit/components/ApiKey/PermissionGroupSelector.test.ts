import { createLocalVue, shallowMount, Stubs } from '@vue/test-utils'
import VueJSModal from 'vue-js-modal'
import Vuex from 'vuex'

import { Modal, Transition } from 'test/unit/stubs'

import PermissionGroupSelector from '@/components/ApiKey/PermissionGroupSelector.vue'
import { PERMISSION_GROUPS, GROUPED_ABILITIES } from '@/components/ApiKey/data'
import auth, { getInitialState as getInitialAuthState } from '@/store/modules/auth'
import { Ability } from '@/store/types'

const localVue = createLocalVue()
localVue.use(VueJSModal)
localVue.use(Vuex)
localVue.directive('tooltip', () => {})

const stubs: Stubs = {
  'check-box': true,
  Modal,
  Transition
}

const actions = Object.values(GROUPED_ABILITIES).reduce((a, b) => a.concat(b))

// TODO: List all abilities here
const abilities: Ability[] = [{ subject: 'all', conditions: {}, actions }]

const newStore = () => {
  const store = new Vuex.Store({
    modules: {
      auth: { ...auth, state: getInitialAuthState() }
    }
  })
  jest.spyOn(store, 'dispatch').mockResolvedValue({})
  return store
}

PERMISSION_GROUPS.forEach(group => {
  it(`matches snapshot for ${group.name}`, async () => {
    const propsData = { group }
    const store = newStore()
    store.commit('auth/SET_ABILITIES', abilities)
    const wrapper = shallowMount(PermissionGroupSelector, {
      localVue, propsData, store, stubs
    })
    wrapper.find('.permission-group__header__collapse-expand').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
})

it('toggles all group abilities when toggling group', async () => {
  const group = PERMISSION_GROUPS[0]
  const propsData = { group }
  const store = newStore()
  store.commit('auth/SET_ABILITIES', abilities)
  const wrapper = shallowMount(PermissionGroupSelector, {
    localVue, propsData, store, stubs
  })

  // we are dealing with manual stubs, so we check attributes rather than props

  wrapper.findAll('check-box-stub').at(0).vm.$emit('change', { checked: true })
  await wrapper.vm.$nextTick()
  wrapper
    .findAll('check-box-stub')
    .wrappers.forEach(w => expect(w.attributes('value')).toEqual('true'))
  await wrapper.vm.$nextTick()

  wrapper.findAll('check-box-stub').at(0).vm.$emit('change', { checked: false })
  await wrapper.vm.$nextTick()

  // attributes set to false come out as undefined
  wrapper
    .findAll('check-box-stub')
    .wrappers.forEach(w => expect(w.attributes('value')).toBeUndefined())
})

it('emits on group toggle', async () => {
  const group = PERMISSION_GROUPS.find(g => g.id === 'manage_team_members')
  const propsData = { group }
  const store = newStore()
  const listener = jest.fn()
  store.commit('auth/SET_ABILITIES', abilities)
  const wrapper = shallowMount(PermissionGroupSelector, {
    localVue, propsData, store, stubs, listeners: { change: listener }
  })

  // we are dealing with manual stubs, so we check attributes rather than props

  wrapper.findAll('check-box-stub').at(0).vm.$emit('change', { checked: true })
  await wrapper.vm.$nextTick()
  expect(listener).toHaveBeenCalledWith('manage_team_members', GROUPED_ABILITIES.manage_team_members)
})

it('emits on ability toggle', async () => {
  const group = PERMISSION_GROUPS.find(g => g.id === 'manage_team_members')
  const propsData = { group }
  const store = newStore()
  const listener = jest.fn()
  store.commit('auth/SET_ABILITIES', abilities)
  const wrapper = shallowMount(PermissionGroupSelector, {
    localVue, propsData, store, stubs, listeners: { change: listener }
  })

  wrapper.find('.permission-group__header__collapse-expand').trigger('click')
  await wrapper.vm.$nextTick()

  wrapper.find('check-box-stub[id=view_team_members]').vm.$emit('change', { checked: true })
  await wrapper.vm.$nextTick()
  expect(listener).toHaveBeenCalledWith('manage_team_members', ['view_team_members'])

  wrapper.find('check-box-stub[id=view_invitations]').vm.$emit('change', { checked: true })
  await wrapper.vm.$nextTick()
  expect(listener).toHaveBeenCalledWith('manage_team_members', ['view_team_members', 'view_invitations'])

  wrapper.find('check-box-stub[id=view_team_members]').vm.$emit('change', { checked: false })
  await wrapper.vm.$nextTick()
  expect(listener).toHaveBeenCalledWith('manage_team_members', ['view_invitations'])
})

it('expands and collapses', async () => {
  const group = PERMISSION_GROUPS.find(g => g.id === 'manage_team_members')
  const propsData = { group }
  const store = newStore()
  const listener = jest.fn()
  store.commit('auth/SET_ABILITIES', abilities)
  const wrapper = shallowMount(PermissionGroupSelector, {
    localVue, propsData, store, stubs, listeners: { change: listener }
  })

  wrapper.find('.permission-group__header__collapse-expand').trigger('click')
  await wrapper.vm.$nextTick()
  expect(wrapper.vm.$data.expanded).toBe(true)
  expect(wrapper.find('.permission-group--expanded').exists()).toBe(true)

  wrapper.find('.permission-group__header__collapse-expand').trigger('click')
  await wrapper.vm.$nextTick()
  expect(wrapper.vm.$data.expanded).toBe(false)
  expect(wrapper.find('.permission-group--expanded').exists()).toBe(false)
})
