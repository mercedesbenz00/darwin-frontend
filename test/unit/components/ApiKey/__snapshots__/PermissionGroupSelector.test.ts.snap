// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for Manage Comments 1`] = `
<div class="permission-group permission-group--expanded">
  <div class="permission-group__header">
    <check-box-stub name="manage_comments" id="manage_comments" label="Manage Comments" type="rect" size="default" class="permission-group__header__control"></check-box-stub> <button class="permission-group__header__collapse-expand">
      <div class="permission-group__header__collapse-expand__text">
        0 / 7 selected
      </div> <img src="/static/imgs/chevron_bottom.svg" class="permission-group__header__collapse-expand__icon">
    </button>
  </div>
  <transition-stub name="toggle">
    <ul class="permission-group__abilities">
      <li class="permission-group__abilities__ability">
        <check-box-stub name="create_comment_thread" id="create_comment_thread" label="Start new comment threads." type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="create_comment" id="create_comment" label="Reply to comment threads" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="delete_comment_thread" id="delete_comment_thread" label="Delete comment threads" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="delete_comment" id="delete_comment" label="Delete comments" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_comment_thread" id="update_comment_thread" label="Edit comment threads" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_comment" id="update_comment" label="Edit comments" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_comment_threads" id="view_comment_threads" label="View comment threads" type="rect" size="default"></check-box-stub>
      </li>
    </ul>
  </transition-stub>
</div>
`;

exports[`matches snapshot for Manage Datasets 1`] = `
<div class="permission-group permission-group--expanded">
  <div class="permission-group__header">
    <check-box-stub name="manage_datasets" id="manage_datasets" label="Manage Datasets" type="rect" size="default" class="permission-group__header__control"></check-box-stub> <button class="permission-group__header__collapse-expand">
      <div class="permission-group__header__collapse-expand__text">
        0 / 20 selected
      </div> <img src="/static/imgs/chevron_bottom.svg" class="permission-group__header__collapse-expand__icon">
    </button>
  </div>
  <transition-stub name="toggle">
    <ul class="permission-group__abilities">
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_datasets" id="view_datasets" label="View datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="export_dataset" id="export_dataset" label="Export datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="create_dataset" id="create_dataset" label="Create datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="fork_dataset" id="fork_dataset" label="Fork datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_dataset" id="update_dataset" label="Update datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="archive_dataset" id="archive_dataset" label="Archive datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_dataset_data" id="update_dataset_data" label="Add data to datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="archive_dataset_items" id="archive_dataset_items" label="Archive and restore items" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="delete_dataset_items" id="delete_dataset_items" label="Delete dataset items" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_annotations" id="view_annotations" label="View annotations" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_annotation_report" id="view_annotation_report" label="View annotation reports" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_dataset_annotators" id="view_dataset_annotators" label="View dataset annotators" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_dataset_exports" id="view_dataset_exports" label="View exported datasets" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_dataset_performance_report" id="view_dataset_performance_report" label="View dataset performance reports" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_dataset_report" id="view_dataset_report" label="View dataset reports" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_annotation_classes" id="view_annotation_classes" label="View annotation classes" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="create_annotation_class" id="create_annotation_class" label="Create annotation classes" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_annotation_class" id="update_annotation_class" label="Update annotation classes" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="delete_annotation_class" id="delete_annotation_class" label="Delete annotation classes" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="import_annotations" id="import_annotations" label="Import annotations" type="rect" size="default"></check-box-stub>
      </li>
    </ul>
  </transition-stub>
</div>
`;

exports[`matches snapshot for Manage Models 1`] = `
<div class="permission-group permission-group--expanded">
  <div class="permission-group__header">
    <check-box-stub name="manage_models" id="manage_models" label="Manage Models" type="rect" size="default" class="permission-group__header__control"></check-box-stub> <button class="permission-group__header__collapse-expand">
      <div class="permission-group__header__collapse-expand__text">
        0 / 4 selected
      </div> <img src="/static/imgs/chevron_bottom.svg" class="permission-group__header__collapse-expand__icon">
    </button>
  </div>
  <transition-stub name="toggle">
    <ul class="permission-group__abilities">
      <li class="permission-group__abilities__ability">
        <check-box-stub name="deploy_model" id="deploy_model" label="Deploy existing neural models" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="train_models" id="train_models" label="Train new neural models" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_models" id="view_models" label="View neural models" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="run_inference" id="run_inference" label="Run model inference" type="rect" size="default"></check-box-stub>
      </li>
    </ul>
  </transition-stub>
</div>
`;

exports[`matches snapshot for Manage Tasks 1`] = `
<div class="permission-group permission-group--expanded">
  <div class="permission-group__header">
    <check-box-stub name="manage_tasks" id="manage_tasks" label="Manage Tasks" type="rect" size="default" class="permission-group__header__control"></check-box-stub> <button class="permission-group__header__collapse-expand">
      <div class="permission-group__header__collapse-expand__text">
        0 / 3 selected
      </div> <img src="/static/imgs/chevron_bottom.svg" class="permission-group__header__collapse-expand__icon">
    </button>
  </div>
  <transition-stub name="toggle">
    <ul class="permission-group__abilities">
      <li class="permission-group__abilities__ability">
        <check-box-stub name="assign_items" id="assign_items" label="Assign items" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_stage" id="view_stage" label="View workflow stages" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_stage" id="update_stage" label="Update item workflow" type="rect" size="default"></check-box-stub>
      </li>
    </ul>
  </transition-stub>
</div>
`;

exports[`matches snapshot for Manage Team 1`] = `
<div class="permission-group permission-group--expanded">
  <div class="permission-group__header">
    <check-box-stub name="manage_team" id="manage_team" label="Manage Team" type="rect" size="default" class="permission-group__header__control"></check-box-stub> <button class="permission-group__header__collapse-expand">
      <div class="permission-group__header__collapse-expand__text">
        0 / 6 selected
      </div> <img src="/static/imgs/chevron_bottom.svg" class="permission-group__header__collapse-expand__icon">
    </button>
  </div>
  <transition-stub name="toggle">
    <ul class="permission-group__abilities">
      <li class="permission-group__abilities__ability">
        <check-box-stub name="archive_team" id="archive_team" label="Archive team" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="create_team" id="create_team" label="Create teams" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="transfer_team_ownership" id="transfer_team_ownership" label="Transfer team ownership" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="manage_customer" id="manage_customer" label="View and update billing information" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_team" id="update_team" label="Update team" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_team" id="view_team" label="View team" type="rect" size="default"></check-box-stub>
      </li>
    </ul>
  </transition-stub>
</div>
`;

exports[`matches snapshot for Manage Team Members 1`] = `
<div class="permission-group permission-group--expanded">
  <div class="permission-group__header">
    <check-box-stub name="manage_team_members" id="manage_team_members" label="Manage Team Members" type="rect" size="default" class="permission-group__header__control"></check-box-stub> <button class="permission-group__header__collapse-expand">
      <div class="permission-group__header__collapse-expand__text">
        0 / 6 selected
      </div> <img src="/static/imgs/chevron_bottom.svg" class="permission-group__header__collapse-expand__icon">
    </button>
  </div>
  <transition-stub name="toggle">
    <ul class="permission-group__abilities">
      <li class="permission-group__abilities__ability">
        <check-box-stub name="delete_membership" id="delete_membership" label="Delete memberships" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="manage_admin_invitations" id="manage_admin_invitations" label="Create admin invites" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="manage_invitations" id="manage_invitations" label="Create invites" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="update_membership" id="update_membership" label="Update team members" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_invitations" id="view_invitations" label="View invitations" type="rect" size="default"></check-box-stub>
      </li>
      <li class="permission-group__abilities__ability">
        <check-box-stub name="view_team_members" id="view_team_members" label="View team members" type="rect" size="default"></check-box-stub>
      </li>
    </ul>
  </transition-stub>
</div>
`;
